### ISSUE TITLE
Primary Key Set To BigInteger

### ISSUE THREAD

**Issue by InfosecGene** [robmorgan/phinx#519]
This is a feature request to be able to set the primary auto_incrementing integer to bigint instead of just integer.

Thanks for your work and your contribution.

**Comment by alexbowers-tecmark** [comment]
You can override the primary key

http://docs.phinx.org/en/latest/migrations.html?highlight=incrementing#creating-a-table

**Comment by InfosecGene** [comment]
Thanks for pointing out the documentation. I haven't found a way to make the key a biginteger, primary key, and auto other than to create the table and then issue a straight alter table afterwards. Happy to receive instruction. Thanks.

**Comment by alexbowers-tecmark** [comment]
A quick look showed that something like this works:

It isn't pretty, and there is probably a better way around it, but for now this should help you out.

``` php
<?php

use Phinx\Migration\AbstractMigration;

class Test extends AbstractMigration
{
    /**
     * Change Method.
     *
     * More information on this method is available here:
     * http://docs.phinx.org/en/latest/migrations.html#the-change-method
     *
     * Uncomment this method if you would like to use it.
     *
    public function change()
    {
    }
    */
    
    /**
     * Migrate Up.
     */
    public function up()
    {
        $table = $this->table('new_table');
        $table->save();
        
        $table->changeColumn('id', 'biginteger');
        $table->save();
    }

    /**
     * Migrate Down.
     */
    public function down()
    {

    }
}
```

**Comment by nicofieri** [comment]
hi, i use changeColumn to set "biginteger" on "id" field, but the "AUTO_INCREMENT" is deleting

**Comment by rbalik** [comment]
Anybody ever figure out a way to do this? I'm losing the AUTO_INCREMENT as well.

**Comment by robmorgan** [comment]
@InfosecGene @rbalik - heres a solution:

``` php
<?php

use Phinx\Migration\AbstractMigration;
use Phinx\Db\Table\Column;

class CreateTable extends AbstractMigration
{
    public function change()
    {
        $column = new Column();
        $column->setName('id')
               ->setType('biginteger')
               ->setIdentity(true);

        $options = array(
            'id'          => false,
            'primary_key' => 'id'
        );

        $table = $this->table('new_table', $options);
        $table->addColumn($column)
              ->create();
    }
}
```

**Comment by flanth** [comment]
With PostgreSQL, the above code still creates an integer column (Phinx v0.5.1).

````
testdb=# \d new_table
                         Table "public.new_table"
 Column |  Type   |                       Modifiers                        
--------+---------+--------------------------------------------------------
 id     | integer | not null default nextval('new_table_id_seq'::regclass)
Indexes:
    "new_table_pkey" PRIMARY KEY, btree (id)
```

Interestingly it produces the correct result with MySQL:

```
mysql> describe new_table;
+-------+------------+------+-----+---------+----------------+
| Field | Type       | Null | Key | Default | Extra          |
+-------+------------+------+-----+---------+----------------+
| id    | bigint(20) | NO   | PRI | NULL    | auto_increment |
+-------+------------+------+-----+---------+----------------+
```

**Comment by flanth** [comment]
A solution to make the above code work would be to change one line in the function getColumnSqlDefinition(Column $column) in [PostgresAdapter.php](https://github.com/robmorgan/phinx/blob/0.5.x-dev/src/Phinx/Db/Adapter/PostgresAdapter.php) from
```php
$buffer[] = 'SERIAL';
```
to
```php
$buffer[] = $column->getType() == 'biginteger' ? 'BIGSERIAL' : 'SERIAL';
```
(currently [line 916](https://github.com/robmorgan/phinx/blob/0.5.x-dev/src/Phinx/Db/Adapter/PostgresAdapter.php#L916))

**Comment by nbsoftware** [comment]
I can confirm the last change (changing line 916 in [PostgresAdapter.php](https://github.com/robmorgan/phinx/blob/0.5.x-dev/src/Phinx/Db/Adapter/PostgresAdapter.php)) works.
Any chance to integrate this fix into a tagged release?
Thanks

**Comment by robmorgan** [comment]
@nbsoftware fixed in cf99c8b, will go in next release.
